### DISCUSSION TITLE
Set up basic HTTP Authentication

### DISCUSSION THREAD

**Discussion by ritchiea** [vercel/next.js#16823] [Help] [answered]
I would like to set up basic HTTP authentication for our staging site so the public cannot accidentally access it. Is there a built in way to do that?

**Comment by c0b41** [comment]
are you using vercel or somethin else?

if vercel then checkout this https://vercel.com/blog/protecting-deployments

**Reply by ritchiea** [reply]
I'm using Heroku.

**Comment by ritchiea** [comment, marked as answer]
If anyone else has this issue in the future and ends up here: I ended up [writing a custom server](https://nextjs.org/docs/advanced-features/custom-server) and using the [http-auth](https://www.npmjs.com/package/http-auth) NPM package.
